infrastructure/src/main/java/org/corfudb/infrastructure/CompactorService.java

```diff
@@ -0,0 +1,163 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import lombok.NonNull;
+import lombok.Setter;
+import org.corfudb.runtime.CorfuCompactorManagement.CheckpointingStatus;
+import org.corfudb.runtime.CorfuCompactorManagement.CheckpointingStatus.StatusType;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.CorfuStoreMetadata.TableName;
+import org.corfudb.runtime.DistributedCompactor;
+import org.corfudb.runtime.collections.CorfuStore;
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.view.Layout;
+import org.corfudb.util.concurrent.SingletonResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+/**
+ * Orchestrates distributed compaction
+ * <p>
+ * Created by Sundar Sridharan on 3/2/22.
+ */
+public class CompactorService implements ManagementService {
+
+    private long compactionTriggerFreqMs = TimeUnit.MINUTES.toMillis(8);
```

Comment:
This is set later based on the runtime parameter. And it's not possible to set it in the constructor because the runtime should not be initialised at this point. Runtime is bootstrapped by the ManagementAgent which inits CompactorService and doing a getRuntime will cause a cyclic dependency.